nle/env/tasks.py

```diff
@@ -262,32 +250,21 @@ def reset(self, *args, **kwargs):
         self.dungeon_explored = {}
         return super().reset(*args, **kwargs)
 
-    def _reward_fn(self, last_response, response, end_status):
+    def _reward_fn(self, last_observation, observation, end_status):
         del end_status  # Unused
 
         reward = 0
-        internal = response.Internal()
-        if internal:
-            obs = response.Observation()
-            s = response.Blstats()
-            if obs and s:
-                glyphs = (
-                    obs.Glyphs()
-                    .DataAsNumpy()
-                    .view(np.int16)
-                    .reshape(base.DUNGEON_SHAPE)
-                )
-                dlevel = response.You().Uz(nle.fbs.DLevel.DLevel())
-                dungeon_num = dlevel.Dnum()
-                dungeon_level = dlevel.Dlevel()
-
-                key = (dungeon_num, dungeon_level)
-                if glyphs is not None:
-                    explored = (glyphs != 0).sum()
-                    explored_old = 0
-                    if key in self.dungeon_explored:
-                        explored_old = self.dungeon_explored[key]
-                    reward = explored - explored_old
-                    self.dungeon_explored[key] = explored
-        time_penalty = self._get_time_penalty(last_response, response)
+        glyphs = observation[self._glyph_index]
+        blstats = observation[self._blstats_index]
+
+        dungeon_num, dungeon_level = blstats[23:25]
+
+        key = (dungeon_num, dungeon_level)
+        explored = np.sum(glyphs != 0)
+        explored_old = 0
+        if key in self.dungeon_explored:
+            explored_old = self.dungeon_explored[key]
+            reward = explored - explored_old
+            self.dungeon_explored[key] = explored
```

Comment:
did you mean to keep these two lines under this conditional block?

Reply:
Good catch, thanks!